testsuite/src/run.rs

```diff
@@ -0,0 +1,42 @@
+use parity_wasm::elements::{deserialize_buffer, serialize, Module};
+use wast::{
+	parser::{parse, ParseBuffer},
+	QuoteModule, Wast, WastDirective,
+};
+
+pub fn check(path: &str) {
+	let path = path.strip_prefix("testsuite/").unwrap();
+	let source = std::fs::read_to_string(path).unwrap();
+	let buffer = ParseBuffer::new(&source).unwrap();
+	let wast = parse::<Wast>(&buffer).unwrap();
+	for kind in wast.directives {
+		match kind {
+			WastDirective::Module(mut module) => {
+				let (line, _col) = module.span.linecol_in(&source);
+				println!("Parsing module at line {}", line);
+				let orig_bytes = module.encode().unwrap();
+				let parsed =
+					deserialize_buffer::<Module>(&orig_bytes).expect("Failed to parse module");
+				serialize(parsed).expect("Failed to serialize module");
+			},
+			WastDirective::AssertMalformed { module, message, span }
+				if matches!(module, QuoteModule::Module(_)) =>
```

Comment:
Can you fold this if-guard into the pattern, i.e. something like `WastDirective::AssertMalformed { module: QuoteModule::Module(_), message, span }`?

Reply:
Yes. I feel stupid now hehe